crates/re_viewport/src/viewport.rs

```diff
@@ -552,9 +550,14 @@ impl TabWidget {
         ui.painter()
             .rect(self.rect, 0.0, self.bg_color, egui::Stroke::NONE);
 
-        let icon_image =
-            egui::widgets::Image::new(self.texture_id, self.icon_size).tint(self.text_color);
-        icon_image.paint_at(ui, self.icon_rect);
+        let icon_image = self
+            .icon
+            .as_image()
+            .fit_to_exact_size(self.icon_size)
+            .tint(self.text_color);
+        if let Ok(TexturePoll::Ready { texture }) = icon_image.load(ui) {
+            icon_image.paint_at(ui, self.icon_rect, &texture);
+        }
```

Comment:
We do this pattern a lot now. We should make sure something like this works instead:

```suggestion
        icon_image.paint_at(ui, self.icon_rect);
```

I feel like everytime we as users see a naked `TexturePoll`, that's a failure in the egui interface

Reply:
(we can save this for later)